components/movementsensor/dualgps/dualgps.go

```diff
@@ -0,0 +1,236 @@
+// gps movement sensors
+package dualgps
+
+import (
+	"context"
+	"math"
+	"sync"
+
+	"github.com/golang/geo/r3"
+	geo "github.com/kellydunn/golang-geo"
+
+	"go.viam.com/rdk/components/movementsensor"
+	"go.viam.com/rdk/logging"
+	"go.viam.com/rdk/resource"
+	"go.viam.com/rdk/spatialmath"
+	"go.viam.com/rdk/utils"
+)
+
+// the default offset between the two gps devices describes a setup where
+// one gps is mounted on the right side of the base and one gps is mounted on the left side of the base
+//   ___________
+//  |	base	|
+//  |			|
+// GPS2			GPS1
+//  |			|
+//  |___________|
+
+const defaultOffsetDegrees = 90.0
+
+var model = resource.DefaultModelFamily.WithModel("dual-gps-rtk")
+
+// Config is used for converting fake movementsensor attributes.
+type Config struct {
+	Gps1   string  `json:"first_gps"`
+	Gps2   string  `json:"second_gps"`
+	Offset float64 `json:"offset_degrees,omitempty"`
+}
+
+// Validate validates the dual gps model's config to make sure that it has
+// two gps movement sensors.
+func (c *Config) Validate(path string) ([]string, error) {
+	var deps []string
+
+	if c.Gps1 == "" {
+		return nil, resource.NewConfigValidationFieldRequiredError(path, "first_gps")
+	}
+	deps = append(deps, c.Gps1)
+
+	if c.Gps2 == "" {
+		return nil, resource.NewConfigValidationFieldRequiredError(path, "second_gps")
+	}
+	deps = append(deps, c.Gps2)
+
+	return deps, nil
+}
+
+func init() {
+	resource.RegisterComponent(
+		movementsensor.API,
+		model,
+		resource.Registration[movementsensor.MovementSensor, *Config]{Constructor: newDualGPS})
+}
+
+type dualGPS struct {
+	resource.Named
+	logger logging.Logger
+	mu     sync.Mutex
+	offset float64
+
+	gps1 gpsWithLock
+	gps2 gpsWithLock
+}
+
+// newDualGPS makes a new fake movement sensor.
+func newDualGPS(ctx context.Context, deps resource.Dependencies, conf resource.Config, logger logging.Logger,
+) (movementsensor.MovementSensor, error) {
+	dg := dualGPS{
+		Named:  conf.ResourceName().AsNamed(),
+		logger: logger,
+	}
+
+	if err := dg.Reconfigure(ctx, deps, conf); err != nil {
+		return nil, err
+	}
+	return &dg, nil
+}
+
+func (dg *dualGPS) Reconfigure(ctx context.Context, deps resource.Dependencies, conf resource.Config) error {
+	newConf, err := resource.NativeConfig[*Config](conf)
+	if err != nil {
+		return err
+	}
+
+	dg.mu.Lock()
+	defer dg.mu.Unlock()
+
+	first, err := movementsensor.FromDependencies(deps, newConf.Gps1)
+	if err != nil {
+		return err
+	}
+	dg.gps1.gps = first
+
+	second, err := movementsensor.FromDependencies(deps, newConf.Gps2)
+	if err != nil {
+		return err
+	}
+	dg.gps2.gps = second
+
+	dg.offset = defaultOffsetDegrees
+	if newConf.Offset != 0 {
+		dg.offset = newConf.Offset
+	}
+
+	return nil
+}
+
+type gpsWithLock struct {
+	gps     movementsensor.MovementSensor
+	gpsLock sync.Mutex
+}
+
+func (lgps *gpsWithLock) getPosition(ctx context.Context, extra map[string]interface{}) (*geo.Point, error) {
+	lgps.gpsLock.Lock()
+	defer lgps.gpsLock.Unlock()
```

Comment:
This mutex also appears to only be used in this one place, preventing us from getting the position twice at the same time. Why is it here at all? All `MovementSensors` are already supposed to be thread safe, so I don't see what this protects against. Consider removing the mutex.

...and at that point, the entire `gpsWithLock` is just a wrapper around `MovementSensor.Position`, so we can probably inline the whole thing and get rid of this struct entirely.